pkg/config/raw_secrets.go

```diff
@@ -0,0 +1,89 @@
+package config
+
+import (
+	"fmt"
+	"path/filepath"
+)
+
+type rawSecret struct {
+	Id         string `yaml:"id"`
+	Env        string `yaml:"env,omitempty"`
+	Src        string `yaml:"src,omitempty"`
+	PlainValue string `yaml:"value,omitempty"`
+
+	doc *doc `yaml:"-"` // parent
+
+	UnsupportedAttributes map[string]interface{} `yaml:",inline"`
+}
+
+func (s *rawSecret) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	parentStack.Push(s)
+	type plain rawSecret
+	err := unmarshal((*plain)(s))
+	parentStack.Pop()
+	if err != nil {
+		return fmt.Errorf("secrets parsing error: %w", err)
+	}
+
+	if err := s.validate(); err != nil {
+		return fmt.Errorf("secrets validation error: %w", err)
+	}
+
+	if err := checkOverflow(s.UnsupportedAttributes, nil, s.doc); err != nil {
+		return fmt.Errorf("secrets validation error: %w", err)
+	}
+
+	return nil
+}
+
+func (s *rawSecret) validate() error {
+	switch {
+	case s.Env != "":
+		if s.Src != "" || s.PlainValue != "" {
```

Comment:
You can use something like that https://github.com/werf/werf/blob/e15cf3ce7ce16de984856e651e87067367e55b3e/pkg/config/common.go#L65 to check it in one place.

Reply:
https://github.com/werf/werf/blob/e15cf3ce7ce16de984856e651e87067367e55b3e/pkg/config/git_remote_export.go#L26-L28

Reply:
done